### PR TITLE
feat: mid event callback to use page (CMC-801) 

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[CMC-801](https://tools.hmcts.net/jira/browse/CMC-801)

### Change description ###
This PR adds page id for mid event callbacks.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by swalker125 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandler.java [hunk 1]
```diff
@@ -36,13 +44,14 @@ public void register(Map<String, CallbackHandler> handlers) {
     }
 
     public CallbackResponse handle(CallbackParams callbackParams) {
-        return ofNullable(callbacks().get(callbackParams.getType()))
+        String callbackKey = callbackKey(callbackParams.getType(), callbackParams.getPageId());
+        return ofNullable(callbacks().get(callbackKey))
             .map(callback -> callback.execute(callbackParams))
             .orElseThrow(() -> new CallbackException(
                 String.format(
-                    "Callback for event %s, type %s not implemented",
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandler.java:43]: will look odd when page id null

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandler.java:43]: So what is the suggestion?

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/AcknowledgeServiceCallbackHandler.java [hunk 2]
```diff
@@ -40,12 +43,12 @@
...
     @Override
-    protected Map<CallbackType, Callback> callbacks() {
+    protected Map<String, Callback> callbacks() {
         return Map.of(
-            CallbackType.ABOUT_TO_START, this::emptyCallbackResponse,
-            CallbackType.MID, this::validateDateOfBirth,
-            CallbackType.ABOUT_TO_SUBMIT, this::setNewResponseDeadline,
-            CallbackType.SUBMITTED, this::buildConfirmation
+            callbackKey(ABOUT_TO_START), this::emptyCallbackResponse,
+            callbackKey(MID, "confirm-details"), this::validateDateOfBirth,
```
Comment (pd-dudkiewicz) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/AcknowledgeServiceCallbackHandler.java:50]: how about keeping all these page ids in an enum?

**Review by pd-dudkiewicz — DISMISSED** [review]
one minor comment, other than that LGTM

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/AcknowledgeServiceCallbackHandler.java (hunk 2 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/AcknowledgeServiceCallbackHandler.java:50]: enum is not going to add any value here so will keep string.

**Review by dharmendrak — APPROVED** [review]

**Review by douglasrice — APPROVED** [review]

**Pull request merged**